src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -1323,6 +1323,45 @@ public void removeChannel(ChannelShim channelShim)
         }
     }
 
+    /**
+     * Update media direction of {@link ChannelShim}s associated
+     * with this Endpoint.
+     *
+     * When media direction is set to 'sendrecv' JVB will
+     * accept incoming media from endpoint and forward it to
+     * other endpoints in a conference. Other endpoint's media
+     * will also be forwarded to current endpoint.
+     * When media direction is set to 'sendonly' JVB will
+     * NOT accept incoming media from this endpoint (not yet implemented), but
+     * media from other endpoints will be forwarded to this endpoint.
+     * When media direction is set to 'recvonly' JVB will
+     * accept incoming media from this endpoint, but will not forward
+     * other endpoint's media to this endpoint.
+     * When media direction is set to 'inactive' JVB will
+     * neither accept incoming media nor forward media from other endpoints.
+     *
+     * @param type media type.
+     * @param direction desired media direction:
+     *                       'sendrecv', 'sendonly', 'recvonly', 'inactive'
+     */
+    public void updateMediaDirection(MediaType type, String direction) {
```

Comment:
What do you see the interactions with this code looking like? (As in, who would call it?)

Reply:
@bbaldino I use jvb as a library for an application which basically implement signaling for jvb (I think it does similar things as jicofo does). So I just use public API of Videobridge to configure desired behavior. I think in my signaling code I'll use  `updateMediaDirection` in following way:
1. To enable hold
```
localEndpoint.updateMediaDirection(MediaType.VIDEO, "inactive");
localEndpoint.updateMediaDirection(MediaType.AUDIO, "inactive");
```
2. To disable video when client is minimized:
```
localEndpoint.updateMediaDirection(MediaType.VIDEO, "inactive");
```

Reply:
Gotcha.  And, with the changes to IO activity to take ICE into account, there's no issues with expiration when a caller is put on hold?

Reply:
Yes, exactly! Endpoint is still treated as connected because there is ICE activity during hold and hence endpoint is not expired.
Fix for IO activity was previoulsy done in https://github.com/jitsi/jitsi-videobridge/pull/970